packages/lab/src/Flow/stories/SubFlow/NodeGroup.tsx

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useState } from "react";
+import { css, cx } from "@emotion/css";
+import {
+  NodeResizeControl,
+  NodeResizer,
+  NodeProps as ReactFlowNodeProps,
+} from "reactflow";
+import {
+  HvDropDownMenu,
+  HvListValue,
+  HvTypography,
+  theme,
+} from "@hitachivantara/uikit-react-core";
+import { Fullscreen } from "@hitachivantara/uikit-react-icons";
+import { useHvNode } from "@hitachivantara/uikit-react-lab";
+
+const dashedBorder = `1px dashed ${theme.colors.atmo4}`;
+const containerBorderRadius = "0 16px 16px 16px";
+const minWidth = 300;
+const minHeight = 200;
+
+const classes = {
+  root: css({
+    display: "flex",
+    width: "100%",
+    height: "100%",
+    backgroundColor: theme.colors.atmo1,
+    borderRadius: containerBorderRadius,
+    minWidth,
+    minHeight,
+    border: `1px solid transparent`,
+  }),
+  content: css({
+    display: "flex",
+    borderRadius: containerBorderRadius,
+    backgroundColor: theme.colors.primary_20,
+    width: "100%",
+    border: dashedBorder,
+  }),
+  header: css({
+    display: "flex",
+    width: "100%",
+    height: "100%",
+    padding: theme.spacing("2px", "xs", "2px", "xs"),
+    borderRadius: "16px 16px 0 0",
+    border: dashedBorder,
+    borderBottom: "none",
+    flexDirection: "row",
+    alignItems: "center",
+    backgroundColor: theme.colors.primary_20,
+  }),
+  headerBackground: css({
+    display: "flex",
+    position: "absolute",
+    top: "-34px",
+    backgroundColor: theme.colors.atmo1,
+    borderRadius: "16px 16px 0 0",
+    borderBottom: "none",
+  }),
+  title: css({
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+  }),
+  actions: css({
+    marginLeft: "auto",
+  }),
+};
+
+interface NodeProps extends ReactFlowNodeProps {
+  groupId: string;
+  actions?: HvListValue[];
+}
+
+export const NodeGroup = ({ id: idProp, groupId, actions = [] }: NodeProps) => {
+  const { title, icon, id, node, intersections, setNodeParent } = useHvNode({
+    id: idProp,
+    groupId,
+  });
+
+  const [draggingFlag, setDraggingFlag] = useState(false);
+
+  useEffect(() => {
```

Comment:
I had a bit of trouble understanding why this code is needed (same for the `useEffect` inside `Node.tsx`). I think adding a comment would be nice 🙏

Reply:
I agree,  added some comments, tell me what you think